docs/_docs/02_features/debugging.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: collection-browser-doc
+title: Debugging
+category: features
+categories_url: features
+excerpt: Learn how to debug issues with terragrunt and terraform.
+tags: ["DRY", "Use cases", "CLI"]
+order: 220
+nav_title: Documentation
+nav_title_link: /docs/
+---
+
+## Debugging
+
+Terragrunt and Terraform usually play well together in helping you
+write DRY, re-usable infrastructure. But how do we figure out what
+went wrong in the rare case that they _don't_ play well?
+
+Terragrunt provides a debug mode you can access through the `--terragrunt-debug`
+command flag. For example you could use it like this to debug an `apply`
+that's producing unexpected output:
+
+    $ terragrunt --terragrunt-debug apply
```

Comment:
Seeing it written out this way makes me wonder if the argument should be simply `--debug` - the `terragrunt` part seems kind of redundant. But let's not dwell too much on that, I'm ultimately fine with either.

Reply:
I _think_ the idea there was to namespace the flag so as not to confuse it with terraform's own debug flag?

But this kind of begs the question when we run terragrunt in debug mode should we also invoke terraform in the same mode?

Reply:
Namespacing makes sense.

I don't think we should invoke terraform with the mode on. People can use `TF_LOG=debug` to enable terraform debugging.

Reply:
For the sake of this 1-day project I will say _no_, but I can only imagine how annoying it would be to a user to execute `terragrunt --terragrunt-debug apply --debug`

Reply:
I'll chime in to confirm that we namespace all Terragrunt-specific args with `--terragrunt`. That ensures they don't get mixed up with the CLI args meant to be passed to Terraform itself.